src/components/UserActivationPage/index.jsx

```diff
@@ -16,50 +17,39 @@ const USER_ACCOUNT_POLLING_TIMEOUT = 5000;
 
 const UserActivationPage = ({
   match,
-  authentication,
-  userAccount,
-  fetchUserAccount,
 }) => {
+  const user = getAuthenticatedUser();
+  const {
+    username, roles, isActive,
+  } = user;
   const [isPollingUserAccount, setIsPollingUserAccount] = useState(false);
   const enterpriseSlug = useMemo(
     () => match.params.enterpriseSlug,
     [match.params],
   );
   const { addToast } = useContext(ToastsContext);
 
-  useEffect(
-    () => {
-      let timeout;
-      if (authentication?.username && userAccount?.loaded && !userAccount?.isActive) {
-        // user is authenticated and we finished hydrating the full user metadata, but
-        // the user has not verified their email address. we start polling for their user
-        // metadata every USER_ACCOUNT_POLLING_TIMEOUT milliseconds.
-        setIsPollingUserAccount(true);
-        timeout = setTimeout(() => {
-          fetchUserAccount(authentication.username);
-        }, USER_ACCOUNT_POLLING_TIMEOUT);
-      }
-      return () => {
-        if (timeout) {
-          clearInterval(timeout);
-        }
-      };
-    },
-    [authentication, userAccount],
-  );
+  useInterval(() => {
+    if (username && !isActive) {
+      setIsPollingUserAccount(true);
+      hydrateAuthenticatedUser();
+    } else if (isActive) {
+      setIsPollingUserAccount(false);
+    }
+  }, USER_ACCOUNT_POLLING_TIMEOUT);
 
-  if (authentication?.username) {
+  if (username) {
     // user is authenticated, but doesn't have any JWT roles so redirect the user to
     // `:enterpriseSlug/admin/register` to display the proper error message.
-    if (!authentication.roles?.length) {
+    if (!roles?.length) {
       return (
         <Redirect to={`/${enterpriseSlug}/admin/register`} />
       );
     }
 
     // user data is hydrated with a verified email address, so redirect the user
     // to the default page in the Admin Portal.
-    if (userAccount?.loaded && userAccount?.isActive) {
+    if (isActive !== undefined && isActive) {
```

Comment:
Does this work??? I thought an undefined variable in javascript throws an error no matter what you try to compare it to?

Reply:
My normal workaround: (typeof isActive !== 'undefined')

Reply:
I tried it in the console and I can compare it like this, so, I'm guessing yes? I think you're thinking about `typeof`?

Reply:
yeah I think Adrienne is correct, 

```
typeof x
"undefined"


x === undefined
VM570:1 Uncaught ReferenceError: x is not defined
    at <anonymous>:1:1
```

typeof should be the safest optoin generally 

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/typeof#description

Reply:
having said that, maybe the destructure above actually defined isActive to undefined?

Reply:
yup!!

```
const { x } = {}
undefined
if (x) { console.log('x') }
undefined
```

Reply:
Yes, destructuring sets `isActive` to undefined.